### PR TITLE
OktaUserDao

### PR DESCRIPTION
**Pull request by frett**
Create an OktaUserDao supporting the following functions:
- Lookup users
- Create/Update users
- Search for users
- List Groups
- Manage Group membership

Things to do in a followup PRs:
- Support Reactivate/Deactivate users
- Potentially optimize API calls to reduce API request load (not sure how this will currently measure up against Okta limits, plus utilizing the built-in SDK caching mechanism should help)

### CODE REVIEW

**Comment by mattdrees** [comment]
I'm out of the office today and tomorrow and most of thursday. I'll try to review some tomorrow evening and see if I can finish up thursday afternoon.

**Comment by frett** [comment]
That's fine, I've got plenty to work on still

**Comment by frett** [comment]
@mattdrees bump

**Comment by mattdrees** [comment]
> bump

yeah, sorry. Made it as far as 68b4d5d36f09fe0d371c94c4508cbf5fbd2c2605, but need to pause for now. Will try to resume later tonight.

File: okta/src/main/java/org/ccci/idm/user/okta/dao/OktaUserDao.kt [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+    }
+
+    // region Stream Users
+    override fun streamUsers(
+        expression: Expression?,
+        includeDeactivated: Boolean,
+        restrictMaxAllowed: Boolean
+    ): Stream<User> {
+        val search = expression?.toOktaExpression()
+        return okta.listUsers(null, null, null, search, null).stream()
```
Comment (mattdrees) [inline okta/src/main/java/org/ccci/idm/user/okta/dao/OktaUserDao.kt]: really neat that the SDK can return a stream for us

Comment (frett) [reply okta/src/main/java/org/ccci/idm/user/okta/dao/OktaUserDao.kt]: Yeah, it automatically handles paging which is really nice

**Pull request merged**